inspirehep/modules/records/receivers.py

```diff
@@ -412,3 +414,21 @@ def populate_author_count(sender, json, *args, **kwargs):
         if 'supervisor' not in author.get('inspire_roles', [])
     ]
     json['author_count'] = len(authors_excluding_supervisors)
+
+
+def populate_authors_full_name_normalized(sender, json, *args, **kwargs):
+    """Populate the ``authors.full_name.raw`` field of Literature records."""
+    if 'hep.json' not in json.get('$schema'):
+        return
+
+    authors = json.get('authors', [])
+
+    for index, author in enumerate(authors):
+        full_name = author.get('full_name')
+        if full_name:
+            if not isinstance(full_name, unicode):
```

Comment:
This is not Python 3 compliant, it's better to cover these cases for when we make the change.  
Look into `six` library and especially their `text_type`.

Reply:
At this point in the code we shouldn't have to deal with bytes; everything should already be decoded (Unicode sandwich!). You had to add these two lines because your tests are using bytes, but this doesn't happen in real data.

Reply:
@jacquerie From the [build log](https://travis-ci.org/inspirehep/inspire-next/jobs/329388387#L1571) I guess when migrating the `full_name` is not in unicode. 
So, we're violating the "unicode sandwich" rule at our boundaries, which should be either in literature builder or dojson.  

I would say @iulianav can add the "check-unicode-barrier" there and at a later point fix it at our system boundaries.